core/services/blockhashstore/feeder.go

```diff
@@ -63,6 +71,30 @@ type Feeder struct {
 	errsLock      sync.Mutex
 }
 
+func (f *Feeder) StartHeartbeats(ctx context.Context) {
+	if f.heartbeatPeriodTime == 0 {
+		f.lggr.Infow("Not starting heartbeat blockhash using storeEarliest")
+		return
+	}
+	f.lggr.Infow("Starting heartbeat blockhash using storeEarliest")
+	timer := time.NewTimer(time.Duration(f.heartbeatPeriodTime) * time.Second)
+	for {
+		select {
+		case <-timer.C:
```

Comment:
I think this code is pretty hard to test due to the timer aspect. I think we can clean this up a bit and make it more testable by making an interface that can be mocked out in tests appropriately.

```golang
type Timer interface {
	After(d time.Duration) <-chan time.Time
}
```

This can have a mock implementation and a real implementation. For the real implementation:

```golang
type realTimer struct {}

func (r *realTimer) After(d time.Duration) <-chan time.Time {
	return time.After(d)
}
```

For the mock implementation, you can generate a mock using mockery and you can do something like:

```golang
mockTimer.On("After", expectedDuration).Return(func() {
	c := make(chan time.Time)
	close(c)
	return c
})
```

to trigger the select clause that will call `StoreEarliest`.

Reply:
Fixed this